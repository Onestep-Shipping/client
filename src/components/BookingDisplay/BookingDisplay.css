.booking-details-container {
  width: 45%;
  padding: 10px 0px;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  margin-bottom: 50px;
  border: rgb(110, 110, 110) solid 1px;
  padding: 35px 20px 20px 20px;
  border-radius: 10px;
}
