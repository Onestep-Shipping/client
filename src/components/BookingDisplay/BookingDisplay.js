import React from 'react';
import './BookingDisplay.css';
import PropTypes from 'prop-types';
import Utils from '../../utils/Helpers.js';

const BookingDisplay = (props) => {
  const { schedule, quote, fields } = props;
  return (
    <div className="booking-details-container">
      <div className="info-row">
        <div>
          <text className="info-label">From: </text>
          <text className="schedule-info-text-left">{schedule.route.startLocation}</text>
        </div>
        <div>
          <text className="info-label">Date: </text>
          <text className="schedule-info-text">{schedule.startDate}</text>
        </div>
      </div>
      <div className="info-row">
        <div>
          <text className="info-label">To: </text>
          <text className="schedule-info-text-left">{schedule.route.endLocation}</text>
        </div>
        <div>
          <text className="info-label">Date: </text>
          <text className="schedule-info-text">{schedule.endDate}</text>
        </div>
      </div>
      <div className="info-row">
        <text className="info-label">Vessels / Services: </text>
        <text className="schedule-result-text">{schedule.vessels}</text>
      </div>
      {fields > 3 &&
        <div style={{width: '100%'}}>
          <div className="info-row">
            <text className="info-label">Carrier: </text>
            <text className="schedule-result-text">{schedule.route.carrier}</text>
          </div>
          <div className="info-row">
            <div>
              <text className="info-label">Transshipments: </text>
              <text className="schedule-result-text">{schedule.transshipment}</text>
            </div>
            <div>
              <text className="info-label">Transit Time: </text>
              <text className="schedule-result-text">{schedule.transitTime} days</text>
            </div>
          </div>

        </div>
      }
      {fields > 5 &&
        <div style={{width: '100%'}}>
          <h2>Quote</h2>
          <div className="info-row">
            <text className="info-label">Validity: </text>
            <text className="schedule-result-text">{Utils.formatValidity(quote.validity)}</text>
          </div>
          {quote.selling.oceanFreight.map((container, ind) => 
             <div className="info-row" key={ind}>
               <text className="info-label">{ind === 0 && "Ocean Freight (All-in): "}</text>
              <text className="schedule-result-text">{container.containerType} - ${container.price}</text>
            </div>
          )}
          <div className="info-row">
            <text className="info-label">Documentation Fee: </text>
            <text className="schedule-result-text">${quote.selling.docFee}</text>
          </div>
          <div className="info-row">
            <text className="info-label">Administration Fee: </text>
            <text className="schedule-result-text">${quote.selling.adminFee}</text>
          </div>
        </div>
      }
    </div>
  );
};

export default BookingDisplay;

BookingDisplay.propTypes = {
  schedule: PropTypes.object, 
  quote: PropTypes.object, 
  fields: PropTypes.number
};
