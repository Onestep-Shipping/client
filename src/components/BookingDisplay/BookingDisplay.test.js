import BookingDisplay from './BookingDisplay';
import React from 'react';
import { shallow } from 'enzyme';

const closeModal = () => {}

let wrapped = shallow( <BookingDisplay closeModal={closeModal} />);

describe('LoginForm', () => {
  it('should render the LoginForm Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });
});