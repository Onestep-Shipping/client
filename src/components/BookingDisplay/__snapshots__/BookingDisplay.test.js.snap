// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingDisplay should render the BookingDisplay Component correctly 1`] = `
<div
  className="booking-details-container"
>
  <div
    className="info-row"
  >
    <div>
      <span
        className="info-label"
      >
        From: 
      </span>
      <span
        className="schedule-info-text-left"
      >
        Vancouver, BC | CA
      </span>
    </div>
    <div>
      <span
        className="info-label"
      >
        Date: 
      </span>
      <span
        className="schedule-info-text"
      >
        04/21/2020
      </span>
    </div>
  </div>
  <div
    className="info-row"
  >
    <div>
      <span
        className="info-label"
      >
        To: 
      </span>
      <span
        className="schedule-info-text-left"
      >
        Vung Tau | VN
      </span>
    </div>
    <div>
      <span
        className="info-label"
      >
        Date: 
      </span>
      <span
        className="schedule-info-text"
      >
        04/21/2020
      </span>
    </div>
  </div>
  <div
    className="info-row"
  >
    <span
      className="info-label"
    >
      Vessels / Services: 
    </span>
    <span
      className="schedule-result-text"
    >
      YM UPSURGENCE / 043W / PN2
    </span>
  </div>
</div>
`;
